src/averageAges.js

```diff
@@ -1,61 +1,84 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(filterMen);
+  const result = men.reduce(averageAge, 0);
+
+  function filterMen(person) {
+    return century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm';
+  }
+
+  function averageAge(prev, person) {
+    const age = person.died - person.born;
+
+    return prev + age;
+  }
```

Comment:
Reduce call with this callback should be outside `calculateMenAverageAge`. And you can use it in `calculateWomenAverageAge` - thats main point of last comments - please fix it